src/Summoner/Question.hs

```diff
@@ -46,14 +47,14 @@ printQuestion question (def:rest) = do
     putTextLn $ "/" <> restSlash
 printQuestion question [] = T.putStrLn question
 
-choose :: forall a. (Show a, Read a) => Text -> [a] -> IO a
+choose :: Text -> [LicenseName] -> IO LicenseName
```

Comment:
I would like to keep this function polymorphic, that will take a function how to parse the concrete type.

If you want to have a license specific one because you don't want to pass the function explicitly then you should create `chooseLicense = choose parseLicenseName`, but I don't see the problem in writing it down and bring `License` in this module.

Reply:
Agree with @vrom911 here. It's better to have function that takes `(Text -> Maybe a)` function explicitly and not hardcode `LicenseName`.